app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/TravelCertificateHistory.kt

```diff
@@ -0,0 +1,422 @@
+          ) {
+            Icon(
+              imageVector = Icons.Hedvig.Info,
+              contentDescription = stringResource(R.string.REFERRALS_INFO_BUTTON_CONTENT_DESCRIPTION),
+              modifier = Modifier.size(24.dp),
+            )
+          }
+        },
+      ) {
+        Spacer(modifier = Modifier.weight(1f))
+        HedvigInformationSection(
+          title = stringResource(id = R.string.something_went_wrong),
+          buttonText = stringResource(id = R.string.GENERAL_RETRY),
+          onButtonClick = {
+            reload()
+          },
+        )
+        Spacer(modifier = Modifier.weight(1f))
+      }
+    }
+
+    CertificateHistoryUiState.Loading -> {
+      Box(modifier = Modifier.fillMaxSize()) {
+        CircularProgressIndicator(modifier = Modifier.align(Alignment.Center))
+      }
+    }
+
+    is CertificateHistoryUiState.SuccessDownloadingCertificate -> {
+      LaunchedEffect(uiState.downLoadingUri) {
+        uiState.downLoadingUri?.let {
+          onShareTravelCertificate(it)
+        }
+      }
+    }
+
+    is CertificateHistoryUiState.SuccessDownloadingHistory -> {
+      ShowHistory(
+        onIconClick = { showBottomSheet = true },
+        onCertificateClick = onCertificateClick,
+        onContinue = onContinue,
+        navigateUp = navigateUp,
+        historyList = uiState.certificateHistoryList,
+      )
+    }
+  }
+}
+
+@Composable
+private fun ShowHistory(
+  onIconClick: () -> Unit,
+  onCertificateClick: (String) -> Unit,
+  onContinue: () -> Unit,
+  navigateUp: () -> Unit,
+  historyList: List<TravelCertificate>,
+) {
+  HedvigScaffold(
+    navigateUp = navigateUp,
+    modifier = Modifier.clearFocusOnTap(),
+    topAppBarText = stringResource(id = R.string.PROFILE_ROW_TRAVEL_CERTIFICATE),
+    topAppBarActions = {
+      IconButton(
+        onClick = { onIconClick() },
+        modifier = Modifier.size(40.dp),
+      ) {
+        Icon(
+          imageVector = Icons.Hedvig.Info,
+          contentDescription = stringResource(R.string.REFERRALS_INFO_BUTTON_CONTENT_DESCRIPTION),
+          modifier = Modifier.size(24.dp),
+        )
+      }
+    },
+  ) {
+    if (historyList.isEmpty()) {
+      Spacer(modifier = Modifier.weight(1f))
+      ShowInitialInfo()
+    } else {
+      ShowNotEmptyList(
+        list = historyList,
+        onCertificateClick = onCertificateClick,
+      )
+    }
+    Spacer(modifier = Modifier.weight(1f))
+    VectorInfoCard(
+      text = String.format(stringResource(id = R.string.travel_certificate_start_date_info), 45),
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(16.dp, 16.dp, 16.dp, 8.dp),
+    )
+    HedvigSecondaryContainedButton(
+      text = stringResource(R.string.travel_certificate_get_travel_certificate_button),
+      onClick = onContinue,
+      modifier = Modifier.padding(16.dp, 8.dp, 16.dp, 0.dp),
+    )
+    Spacer(Modifier.height(32.dp))
+  }
+}
+
+@Composable
+private fun ShowInitialInfo() {
```

Comment:
```suggestion
private fun NoCertificateHistorySection() {
```
Perhaps? So that we go away from a composable which shows UI from being a verb